### DISCUSSION TITLE
Open Android Settings

### DISCUSSION THREAD

**Discussion by salestv** [airsdk/Adobe-Runtime-Support#2459] [Q&A] [answered]
Hi,

is there any way to open the Android Wi-Fi Settings within a app?

I've looked around and haven't found anything suitable.

**Comment by itlancer** [comment]
@salestv, you can write AIR native extension for it and open settings from Java code https://stackoverflow.com/questions/2318310/how-to-call-wi-fi-settings-screen-from-my-application-using-android

**Comment by salestv** [comment]
I know i could write an ANE but unfortunately i am not familiar with it.

**Comment by marchbold** [comment, marked as answer]
You can do this with our [Share ANE](https://airnativeextensions.com/extension/com.distriqt.Share). It has a function in it to call `startActivity`

```as3
var intent:Intent = new Intent();
intent.action = "android.settings.WIFI_SETTINGS";

Share.service.applications.startActivity( intent );
```

**Reply by marchbold** [reply]
Ref: https://docs.airnativeextensions.com/docs/share/launch-applications#start-actvity

**Comment by salestv** [comment]
Thank you Michael. Works like a charm.

**Reply by marchbold** [reply]
Glad to hear!
